### DISCUSSION TITLE
Change lualine clock to 12 hour time

### DISCUSSION THREAD

**Discussion by Definite-Internet-User** [LazyVim/LazyVim#2528] [Recipes]
I didn't like Lazyvim's default 24 hour clock so I changed it to a 12 hour. Not sure how many other people are like me but hopefully this will help them out.
```
return {
  {
    "nvim-lualine/lualine.nvim",
    opts = {
      sections = {
        lualine_z = {
          function()
            return " " .. os.date("%-I:%-M %p")
          end,
        },
      },
    },
  },
}
```

**Comment by thegreatestgiant** [comment]
Helped me out, the only issue is that it displayed single digit numbers like `1:7` instead of `1:07` so I modified the code

```lua
return {
  {
    "nvim-lualine/lualine.nvim",
    opts = {
      sections = {
        lualine_z = {
          function()
            return " " .. os.date("%-I:%02M %p")
          end,
        },
      },
    },
  },
}
```

**Reply by wulfalpha** [reply]
A small Edit. %r will build the time based on your local. IMO it makes the code look cleaner as it will put the "right" amount of zeros based on local settings.
`return {
  {
    "nvim-lualine/lualine.nvim",
    opts = {
      sections = {
        lualine_z = {
          function()
            return " " .. os.date("%r")
          end,
        },
      },
    },
  },
}
`

**Reply by centuryx476** [reply]
This was EXACTLY what I was looking for.

Thank You

**Reply by luiz-cunha3** [reply]
If anybody stumbles upon this, let me be of help.

You can simply choose how you'd like the time and date to be displayed by looking at the reference [here](https://www.lua.org/pil/22.1.html).

just put the string with the modifiers you want

**Reply by Pstock7** [reply]
@luiz-cunha3 is the reference you linked exhaustive? I do not see `%r` or `%R` anywhere on the page and have been unable to find the documentation for it anywhere

**Reply by dpetka2001** [reply]
@Pstock7 `:h os.date()` mentions
>  If {format} is not `"*t"`, then `date` returns the date as a string,
        formatted according to the same rules as the C function `strftime`.

So, if you then do `<leader>sM` to search the man pages and search for `strftime` you will see all the available format modifiers for the function.
